packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -398,8 +401,11 @@ export function commitPassiveHookEffects(finishedWork: Fiber): void {
       case ForwardRef:
       case SimpleMemoComponent:
       case Chunk: {
-        commitHookEffectList(UnmountPassive, NoHookEffect, finishedWork);
-        commitHookEffectList(NoHookEffect, MountPassive, finishedWork);
+        // TODO (#17945) We should call all passive destroy functions (for all fibers)
+        // before calling any create functions. The current approach only serializes
+        // these for a single fiber.
+        commitHookEffectList(HookPassive, NoHookEffect, finishedWork);
+        commitHookEffectList(NoHookEffect, HookPassive, finishedWork);
```

Comment:
Following from previous comment, here you would combine the tag with `HookHasEffect` before passing them in:

```js
commitHookEffectList(HookPassive | HookHasEffect, NoHookEffect, finishedWork);
commitHookEffectList(NoHookEffect, HookPassive | HookHasEffect, finishedWork);
```

(I assume Closure inlines those values, or you could also hoist them out to module scope)